DisguiseUnityRenderStream/Runtime/PipelineAbstraction/DepthCopy.cs

```diff
@@ -0,0 +1,193 @@
+        {
+            public Shader m_shader;
+            public Material m_material;
+            public int m_pass;
+
+            public bool IsLoaded => m_shader != null && m_material != null && m_pass >= 0;
+
+            public void SetKeyword(string keyword, bool value)
+            {
+                if (IsLoaded)
+                {
+                    if (value)
+                        m_material.EnableKeyword(keyword);
+                    else
+                        m_material.DisableKeyword(keyword);
+                }
+            }
+
+            public void Create(string shaderPath, string shaderPass)
+            {
+                m_shader = Shader.Find(shaderPath);
+                if (m_shader != null)
+                {
+                    m_material = CoreUtils.CreateEngineMaterial(m_shader);
+                    m_pass = m_material.FindPass(shaderPass);
+                }
+            }
+
+            public void Dispose()
+            {
+                CoreUtils.Destroy(m_material);
+            }
+        }
+
+        public Mode mode
+        {
+            get => m_mode;
+            set
+            {
+                m_mode = value;
+                
+                switch (m_mode)
+                {
+                    case Mode.Raw:
+                        m_shaderResources = s_shaderVariantResources.m_Raw;
+                        break;
+                    case Mode.Eye:
+                        m_shaderResources = s_shaderVariantResources.m_Eye;
+                        break;
+                    case Mode.Linear01:
+                        m_shaderResources = s_shaderVariantResources.m_linear01;
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+        }
+        
+        const string k_profilerTag = "Disguise Depth Copy";
+        const string k_ShaderPass = "Depth Copy";
+
+        static ShaderVariantResources s_shaderVariantResources;
+
+        Mode m_mode = Mode.Linear01;
+        ShaderResources m_shaderResources;
+
+        public DepthCopy()
+        {
+            if (!s_shaderVariantResources.IsLoaded)
+            {
+                s_shaderVariantResources.Create(k_ShaderName, k_ShaderPass);
+                Assert.IsTrue(s_shaderVariantResources.IsLoaded, "Couldn't load the shader resources for DepthCopy");
+            }
+
+            mode = Mode.Linear01;
+        }
+
+        /// <summary>
+        /// Performs the copy using the SRP's currently active camera and the provided <see cref="FrameData"/>.
+        /// </summary>
+        public void Execute(ScriptableRenderContext context, FrameData data)
+        {
+            if (!data.IsValid || !m_shaderResources.IsLoaded)
+                return;
+
+            ValidatePipeline();
+            
+            CommandBuffer cmd = CommandBufferPool.Get(k_profilerTag);
+            cmd.Clear();
```

Comment:
you can skip that .Clear I think, the pool handles it, feel free to step in the code to validate it.. Iirc the pool does automatic clearing

Reply:
Nice! The objects are cleared before being added to the free pool. Always been this way it seems.

`static ObjectPool<CommandBuffer> s_BufferPool = new ObjectPool<CommandBuffer>(null, x => x.Clear());`